Kernel/Tracing/ProcessTracer.h

```diff
@@ -61,7 +63,6 @@ class ProcessTracer : public File {
         u32 result;
     };
 
-    pid_t m_pid;
-    bool m_dead { false };
+    Process* m_process;
```

Comment:
There's a problem with using a weak pointer here: weak pointer gets cleared when the pointee is destructed, but `Process` is special because of its multi-step destruction ceremony. Here, we want the process pointer cleared as soon as `Process::die()` is called, which is way before it gets finalized and finally destructed.

Reply:
Ah, touché! That's a really good point. Also lol @ "multi-step destruction ceremony" which is exactly what it is.. :D